src/bin/pgcopydb/archive_iterator.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * src/bin/pgcopydb/archive_iterator.h
+ *   Implementations of a file iterator for reading new lıne seperated files
+ */
+
+#ifndef ARCHIVE_ITER_H
+#define ARCHIVE_ITER_H
+
+#include <stdbool.h>
+
+typedef struct ArchiveIterator ArchiveIterator;
+typedef struct ArchiveContentItem ArchiveContentItem;
+
+ArchiveIterator * archive_iterator_from(const char *filename);
```

Comment:
Have a look around the iterator constructs in catalog.c, and consider having an API that looks more like that.